MapboxCoreNavigation/DistanceFormatter.swift

```diff
@@ -4,38 +4,60 @@ let metersPerMile: CLLocationDistance = 1_609.344
 let secondsPerHour = 60.0 * 60.0
 let yardsPerMile = 1_760.0
 let feetPerMile = yardsPerMile * 3.0
+let feetPerMeter = 3.28084
+
+extension CLLocationDistance {
+    var miles: CLLocationDistance {
+        return self / metersPerMile
+    }
+    
+    var feet: CLLocationDistance {
+        return self * feetPerMeter
+    }
+}
 
 /// Provides appropriately formatted, localized descriptions of linear distances.
 @objc(MBDistanceFormatter)
 public class DistanceFormatter: LengthFormatter {
     /// True to favor brevity over precision.
     var approx: Bool
-    /// True to insert hints for text-to-speech.
-    var forVoiceUse: Bool
+    
     let nonFractionalLengthFormatter = LengthFormatter()
     
+    var usesMetric: Bool {
+        let locale = numberFormatter.locale as NSLocale
+        guard let measurementSystem = locale.object(forKey: .measurementSystem) as? String else {
+            return false
+        }
+        return measurementSystem == "Metric"
+    }
+    
     /**
      Intializes a new `DistanceFormatter`.
      
      - parameter approximate: approximates the distances.
-     - parameter forVoiceUse: If true, the returned string will contain strings for some numbers like fractions. This is helpful for some speech synthesizers.
      */
-    public init(approximate: Bool = false, forVoiceUse: Bool = false) {
+    public init(approximate: Bool = false) {
         self.approx = approximate
-        self.forVoiceUse = forVoiceUse
         super.init()
     }
     
     public required init?(coder decoder: NSCoder) {
         approx = decoder.decodeBool(forKey: "approximate")
-        forVoiceUse = decoder.decodeBool(forKey: "forVoiceUse")
         super.init(coder: decoder)
     }
     
     public override func encode(with aCoder: NSCoder) {
         super.encode(with: aCoder)
         aCoder.encode(approx, forKey: "approximate")
-        aCoder.encode(forVoiceUse, forKey: "forVoiceUse")
+    }
+    
+    func maximumFractionDigits(for distance: CLLocationDistance) -> Int {
+        if usesMetric {
+            return distance < 3_000 ? 1 : 0
+        } else {
+            return distance.miles < 3 ? 1 : 0
```

Comment:
Added a test case to verify that this is not the case https://github.com/mapbox/mapbox-navigation-ios/pull/383/files#diff-e0d6099c4f87ad73208824318f1f72a0R39